indexer/src/candidates_validator.rs

```diff
@@ -26,50 +35,147 @@ impl CandidatesValidator {
         }
     }
 
+    async fn ticker(
+        mut done: mpsc::Receiver<()>,
+        queue_protected: ProtectedQueue,
+        publisher_protected: ProtectedPublisher,
+        view_client: actix::Addr<near_client::ViewClientActor>,
+    ) {
+        let mut interval = time::interval(Duration::from_secs(2));
+
+        loop {
+            tokio::select! {
+                _ = interval.tick() => {
+                    info!(target: "ticker", "trying to flush");
+                    let mut queue = queue_protected.lock().await;
+                    let _ = Self::flush(&mut queue, publisher_protected.clone(), &view_client).await;
+                },
+                _ = done.recv() => {
+                    return
+                }
+            }
+        }
+    }
+
+    // Assumes queue is under mutex
+    async fn flush(
+        queue: &mut VecDeque<CandidateData>,
+        publisher_protected: ProtectedPublisher,
+        view_client: &actix::Addr<near_client::ViewClientActor>,
+    ) -> Result<bool> {
+        while let Some(candidate_data) = queue.front() {
+            let execution_outcome = Self::fetch_execution_outcome(view_client, candidate_data).await?;
+            match execution_outcome.outcome.status {
+                ExecutionStatusView::Unknown | ExecutionStatusView::SuccessReceiptId(_) => return Ok(false),
```

Comment:
What if an execution remains indefinitely with an Unknown` status? We could probably just drop these candidates after some seconds.

Reply:
The status here changed. But it means pending, there's no way i guess it can stay this way and not be executed.